sno/show.py

```diff
@@ -0,0 +1,148 @@
+import contextlib
+import functools
+import json
+from datetime import datetime, timezone, timedelta
+from io import StringIO
+
+import click
+import pygit2
+
+from .cli_util import MutexOption
+from .exceptions import NotFound, NO_COMMIT
+from .timestamps import to_iso8601_utc, to_iso8601_tz
+from . import diff
+
+
+@click.command()
+@click.pass_context
+@click.option(
+    "--text",
+    "output_format",
+    flag_value="text",
+    default=True,
+    help="Show commit in text format",
+    cls=MutexOption,
+    exclusive_with=["json"],
+)
+@click.option(
+    "--json",
+    "--patch",
+    "-p",
+    "output_format",
+    flag_value="json",
+    help="Show commit in JSON patch format",
+    cls=MutexOption,
+    exclusive_with=["text"],
+)
```

Comment:
Discussed out-of-band: going to leave this as-is because:

* we're likely to add HTML output to this command shortly (we already have HTML diff output, so it's just a matter of adding a header with commit information)
* the aliases `-p` and `--patch` are helpful (since JSON output is also the way you generate an applyable patch), so we don't want to remove those